Update to use k8s CRD

rbrndt opened the pull request:
Update to use the beta kubernetes
Custom Resource Definition

rbrndt committed:
Update to use k8s CRD

Update to use the beta kubernetes
Custom Resource Definition

aadams19 commented on pkg/apis/alarmdefinition/v1/types.go:
```diff
@@ -0,0 +1,57 @@
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Are the comments in this file here for a reason?

rbrndt replied:
They are a part of the auto-generation

aadams19 commented on main.go:
```diff
@@ -0,0 +1,436 @@
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/record"
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
```
This is confusing to me. I don't see you using the library anywhere, so how does it work if commented?

rbrndt replied:
That is a leftover from previous iterations, will remove

aadams19 commented on main.go:
```diff
@@ -0,0 +1,436 @@
+						continue
+					}
+					log.Print(err)
+					//applyError(item, err)
```
Remove?

aadams19 commented on main.go:
```diff
@@ -0,0 +1,436 @@
+	}
+	alarmDefinitionCache[result.ID] = *result
+	newDef := convertToADResource(r, result)
+	sampleclientset.Monasca().AlarmDefinitions(*namespace).Update(newDef)
```
Why don't you check for errors in some of your monasca client calls?

rbrndt replied:
this line is a kubernetes CRD client call, but I should check the error

aadams19 commented on main.go:
```diff
@@ -0,0 +1,436 @@
+	if r.Spec.ID == "" {
+		log.Print("No ID")
+		return false
+	} else if r.Spec.ID != existing.ID {
```
just a nit but probably don't need the else here

aadams19 commented on main.go:
```diff
@@ -0,0 +1,436 @@
+		log.Printf("Expression: %s != %s", r.Spec.Expression, existing.Expression)
+		return false
+	}
+	//if r.Spec.Deterministic != existing.Deterministic {
```
remove?

aadams19 commented on README.md:
````diff
@@ -134,3 +134,11 @@ metadata: {}
+Setup:
+1) Ensure https://github.com/kubernetes/code-generator is present in the vendor directory. It may need to be manually added.
+ This is required to run the code generation scripts.
+2) Run ```./hack/update-codegen.sh``` from the main repo directory.
````
Will this update all of the files you have take from k8s?

rbrndt replied:
Yes, it will regenerate all the required k8s code in the pkg directory

aadams19 commented on README.md:
````diff
@@ -134,3 +134,11 @@ metadata: {}
 resourceVersion: ""
 selfLink: ""
 ```
+
````
Are there any changes to the responses above?

rbrndt replied:
Yes, thanks for the reminder

aadams19 commented:
```
# github.com/monasca/alarm-definition-controller
./main.go:94: cannot use prometheusEndpoint (type *string) as type string in argument to http.ListenAndServe
The command '/bin/sh -c glide install &&     go build -a -o ./adc' returned a non-zero code: 2
```

rbrndt committed:
Update from comments

Add deterministic to definitions
Remove extra logging
Update README for new behavior
Use value instead of pointer for prometheus endpoint

rbrndt committed:
Missed prometheus endpoint change

aadams19 commented:
Awesome, that fixed it. Testing now

rbrndt committed:
Add default port to prometheus endpoint
